File: backend/queute/src/main/java/tn/cloudnine/queute/model/roadmap/RoadMapCreatorScore.java
package tn.cloudnine.queute.model.roadmap;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import tn.cloudnine.queute.model.user.User;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class RoadMapCreatorScore {
    private User user;
    private int TotlaNbrRoadMap;
    private int TotaleNbrApprovalls;
    private int TotaleNbrFollow;
    private Long score;
}

File: frontend/src/app/models/RoadMapCreatorScore.ts
import { User } from "./User";

export class RoadMapCreatorScore {
    user: User;
    TotlaNbrRoadMap: number;
    TotaleNbrApprovalls: number;
    TotaleNbrFollow: number;
    score: number; // Using number instead of Long as TypeScript doesn't have Long by default

    constructor(init?: Partial<RoadMapCreatorScore>) {
        this.user = init?.user ?? new User();
        this.TotlaNbrRoadMap = init?.TotlaNbrRoadMap ?? 0;
        this.TotaleNbrApprovalls = init?.TotaleNbrApprovalls ?? 0;
        this.TotaleNbrFollow = init?.TotaleNbrFollow ?? 0;
        this.score = init?.score ?? 0;
    }
}